### DISCUSSION TITLE
dffgf

### DISCUSSION THREAD

**Discussion by TBDD2020** [directus/directus#10397] [Archive]

**Comment by oreilles** [comment]
What release are you on right now, and what is the content of your `directus_migrations` table ?

**Comment by TBDD2020** [comment]
> What release are you on right now, and what is the content of your `directus_migrations` table ? @Oreilles 

I am on 9.0.0.rc95 right now because the 9.0.2 version wont even show me the login page, atleast on the rc95 i can login and get connected. even if the database error is happening. Please see below the database table for directus_migrations.


```
directus-# \dS directus_migrations
                       Table "public.directus_migrations"
  Column   |           Type           | Collation | Nullable |      Default
-----------+--------------------------+-----------+----------+-------------------
 version   | character varying(255)   |           | not null |
 name      | character varying(255)   |           | not null |
 timestamp | timestamp with time zone |           |          | CURRENT_TIMESTAMP
Indexes:
    "directus_migrations_pkey" PRIMARY KEY, btree (version)

```

```
timestamp | 2021-10-07 20:40:07.178925+00
-[ RECORD 29 ]-----------------------------
version   | 20210831A
name      | Remove Limit Column
timestamp | 2021-10-07 20:40:07.183269+00
-[ RECORD 30 ]-----------------------------
version   | 20210903A
name      | Add Auth Provider
timestamp | 2021-10-07 20:40:07.195596+00
-[ RECORD 31 ]-----------------------------
version   | 20210907A
name      | Webhooks Collections Not Null
timestamp | 2021-10-07 20:40:07.205429+00
-[ RECORD 32 ]-----------------------------
version   | 20210910A
name      | Move Module Setup
timestamp | 2021-10-07 20:40:07.213989+00
-[ RECORD 33 ]-----------------------------
version   | 20210920A
name      | Webhooks URL Not Null
timestamp | 2021-10-07 20:40:07.222879+00
-[ RECORD 34 ]-----------------------------
version   | 20210927A
name      | Replace Fields Group
timestamp | 2021-10-07 20:40:07.228844+00
-[ RECORD 35 ]-----------------------------
version   | 20210927B
name      | Replace M2M Interface
timestamp | 2021-10-07 20:40:07.235273+00
-[ RECORD 36 ]-----------------------------
version   | 20210924A
name      | Add Collection Organization
timestamp | 2021-12-08 06:51:16.448489+00
-[ RECORD 37 ]-----------------------------
version   | 20210929A
name      | Rename Login Action
timestamp | 2021-12-08 06:51:16.464151+00
-[ RECORD 38 ]-----------------------------
version   | 20211007A
name      | Update Presets
timestamp | 2021-12-08 06:51:16.495232+00
-[ RECORD 39 ]-----------------------------
version   | 20211009A
name      | Add Auth Data
timestamp | 2021-12-08 06:51:16.498055+00
-[ RECORD 40 ]-----------------------------
version   | 20211016A
name      | Add Webhook Headers
timestamp | 2021-12-08 06:51:16.500638+00
-[ RECORD 41 ]-----------------------------
version   | 20211103A
name      | Set Unique to User Token
timestamp | 2021-12-08 06:51:16.511962+00
-[ RECORD 42 ]-----------------------------
version   | 20211103B
name      | Update Special Geometry
timestamp | 2021-12-08 06:51:16.515932+00
-[ RECORD 43 ]-----------------------------
version   | 20211104A
name      | Remove Collections Listing
timestamp | 2021-12-08 06:51:16.518593+00
-[ RECORD 44 ]-----------------------------
version   | 20211118A
name      | Add Notifications
timestamp | 2021-12-08 10:29:55.839121+00

```

**Comment by oreilles** [comment]
Your database is showing that all migrations have been run all the way up to the latest `9.1.2` release.
Could you try updating your Directus instance to `9.1.2` and tell us what issues you face from there ?

**Comment by oreilles** [comment]
Sorry, the latest version actually is `9.2.1`.

**Comment by TBDD2020** [comment]
@Oreilles updated to 9.2.1 & get a blank screen, no login no sign up in the /admin just blank pages with 

Heres the install

```
/:/var/directus# npm install directus@v9.2.1
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/rollup/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN rollup-plugin-typescript2@0.31.1 requires a peer of typescript@>=2.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN ts-type@2.1.2 requires a peer of @types/bluebird@* but none is installed. You must install peer dependencies yourself.
npm WARN ts-type@2.1.2 requires a peer of ts-toolbelt@^9.6.0 but none is installed. You must install peer dependencies yourself.

+ directus@9.2.1
updated 12 packages and audited 1113 packages in 86.259s

1 package is looking for funding
  run `npm fund` for details

found 6 vulnerabilities (1 moderate, 5 high)
  run `npm audit fix` to fix them, or `npm audit` for details

:/var/directus# npx directus database migrate:latest
13:22:03 ✨ Running migrations...
13:22:03 ✨ Database up to date

```


Heres the Errors in the browser console for /admin page but i see all files have 200 status in the browser console network tab. Although preview of the files is giving me a "We're sorry but Directus doesn't work without JavaScript enabled. Please enable it to continue."

```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
vendor.162b55e9.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
use-sync.6d51d5e5.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
```

Any idea? nginx T file is here: https://pastebin.com/wKSzeu75


And heres the database for directus_migrations after the update:

```
-[ RECORD 40 ]-----------------------------
version   | 20211016A
name      | Add Webhook Headers
timestamp | 2021-12-08 06:51:16.500638+00
-[ RECORD 41 ]-----------------------------
version   | 20211103A
name      | Set Unique to User Token
timestamp | 2021-12-08 06:51:16.511962+00
-[ RECORD 42 ]-----------------------------
version   | 20211103B
name      | Update Special Geometry
timestamp | 2021-12-08 06:51:16.515932+00
-[ RECORD 43 ]-----------------------------
version   | 20211104A
name      | Remove Collections Listing
timestamp | 2021-12-08 06:51:16.518593+00
-[ RECORD 44 ]-----------------------------
version   | 20211118A
name      | Add Notifications
timestamp | 2021-12-08 10:29:55.839121+00
```

**Comment by oreilles** [comment]
Could you try setting up a local instance with `npm init directus-project directus-project`, connect it to your database and see if that works ? That could help us understand if the problem comes from your NGINX / Digitial Ocean configuration or from Directus itself.

**Comment by rijkvanzanten** [comment]
I'll transfer this to a Q&A discussion, as it sounds like there's a infrastructure/networking problem rather than a bug in Directus.

**Reply by TBDD2020** [reply]
but why? the issue of the database having its migration not successfully happen is still happening, the mismatch of the database is still happening.... and what about downgrading back to rc95? and its issue?

**Reply by oreilles** [reply]
We try as much as possible to solve issue related to Directus, but cannot spend as much time providing assistance when the issue likely comes from a server misconfiguration, which appears to be the case here. In order to confirm that hypothesis, have you had a chance to try what I suggested in my previous comment: https://github.com/directus/directus/discussions/10397#discussioncomment-1773477 ?

**Comment by uebmaster** [comment]
Try this, maybe works

https://discord.com/channels/725371605378924594/741330910594465823/836713322639785986

**Reply by TBDD2020** [reply]
No not related to pm2
but thanks for trying

**Comment by yumyo** [comment]
I have exactly the same issue, any help would be really appreciated.

**Reply by rijkvanzanten** [reply]
Which one issue? There's a couple mentioned in the thread above. I think it's best if you could open a new issue detailing exactly what you're running into 🙂

**Reply by yumyo** [reply]
The blank screen after attempting the upgrade. 
After a couple of hours of failed attempts, I ended up backing up the database, destroying the droplet and uploading the database on a new one. Unfortunately, after the dump and import of the DB I faced some permission errors too. 
Luckily in my case, it was not involving much data and it was not hard to recreate where needed. What would be a safe way of backing up / restoring the data in case of an upgrade going wrong? It'd be amazing to see a section in the docs about this critical aspect.

**Reply by rijkvanzanten** [reply]
I can't really comment on what went wrong without more information about the old/new version, database used, server infrastructure etc etc. Generally speaking, the flow is:

- Backup your database and files
- Update the Directus version
- Run the database migrations

**Comment by toby-pondr** [comment]
If anyone is interested, I had the same blank screen issue. I was able to resolve it by shutting down my Directus screen session and starting it again. I hope that helps someone.
